tests/runner.pm

```diff
@@ -1035,12 +1034,12 @@ sub singletest_run {
     if ($torture) {
         $cmdres = torture($CMDLINE,
                           $testnum,
-                          "$gdb --directory $LIBDIR " . shell_quote($DBGCURL) . " -x $LOGDIR/gdbcmd");
+                          "$gdb --directory $LIBDIR " . shell_quote($DBGCURL) . " -x $LOGDIR" . "gdbcmd");
     }
     elsif($gdbthis == 1) {
         # gdb
         my $GDBW = ($gdbxwin) ? "-w" : "";
-        runclient("$gdb --directory $LIBDIR " . shell_quote($DBGCURL) . " $GDBW -x $LOGDIR/gdbcmd");
+        runclient("$gdb --directory $LIBDIR " . shell_quote($DBGCURL) . " $GDBW -x $LOGDIR" . "gdbcmd");
         $cmdres=0; # makes it always continue after a debugged run
```

Comment:
@vszakats what's the thinking here? What's this change for? This completely breaks using runtests `-g`.

Reply:
I made #16495 in the mean time, but there might be something I don't understand here

Reply:
I made $LIBDIR extendable with extra subdirs (for multi-target) via an env. That env should have started with a slash. Eg /Debug. MSYS automagically "corrects" such values to an abs path. So my solution (also to avoid MSYS specific tricks) was to make $LIBPATH end with a slash by default, and pass Debug/ with an ending slash via the env.

But there seem to be cases where LIBDIR is missing an ending slash!

I can do another round to robustify this.

Reply:
Ah, touching $LOGDIR was totally unnecessary here, I'm guessing that breaks it.